### ISSUE TITLE
Watch 

### ISSUE THREAD

**Issue by voiceofwisdom** [fsharp/FAKE#761]
It would be nice to have a `Watch` function, similar to gulps watch.

``` f#
// Pattern specifying assemblies to be tested using NUnit
let testAssemblies = "tests/**/bin/Release/*Tests*.dll"

let testSource = "tests/**/*.fs"
let appSource = "src/**/*.fs"

// File system information 
let solutionFile  = "FSharp.QueryProvider.sln"

Target "Build" (fun _ ->
    !! solutionFile
    |> MSBuildRelease "" "Rebuild"
    |> ignore
)

Target "RunTests" (fun _ ->
    !! testAssemblies
    |> NUnit (fun p ->
        { p with
            DisableShadowCopy = true
            TimeOut = TimeSpan.FromMinutes 20.
            OutputFile = "TestResults.xml" })
)

Target "Watch" (fun _ ->
	// I assume its possible to concat sources, dont know exactly how
	(!! testSource) @ (!! appSource)
    |> Watch (fun changes ->
		//reruns any time the sources change on disk.
		//changes is a list of changed files
    	RunTarget "Build"
    	RunTarget "RunTests" )
)
```

**Comment by VoiceOfWisdom** [comment]
`changes` having the type of seq<FileChange>
``` f#
type FileStatus =
| Created
| Removed
| Changed

type FileChange = {
    Name : string
    Path : string
    Status : FileStatus
}
```

**Comment by voiceofwisdom** [comment]
Closed by pr #766
